RELEASING.md

```diff
@@ -0,0 +1,24 @@
+## Releasing a new version
+
+[This GitHub Action](.github/workflows/release.yaml) builds, and releases to both OSS Maven2 repo and Github a new version of
+the wrapper
+whenever a new tag is pushed to the repository.
+
+Full release procedure is as follows:
+
+* Checkout the latest version of the `main` branch.
+* Update wrapper POM version to release (remove -SNAPSHOT)
+* Set wrapper dependency for examples to release (`examples/build.gradle`)
+* Commit / push / merge PR
+* Tag that version (eg `v0.0.5` - min `v` prefix!) with the [annotated tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) and `git push origin <tagname>` 
```

Comment:
won't this generally be pushing to `upstream` not origin? If people are using standard remote naming?

Reply:
Interesting question. As this is not very well documented I had to test with my private github account. Basically for non-members github won't allow tag push to origin, while members can clone (and push to origin which is the original repo). So IMO we can leave "origin" here.

Reply:
I think what John means here is that if a maintainer follows the standard remote naming, then they will push to `upstream`.

Reply:
Yeah, but maintainer can clone (so no need for upstream, it's origin). Or do we want to keep to "fork" workflow even in such cases? I have the repo cloned on my machine.

Reply:
Whatever we go with, we should clarify what origin is referring to here.

Reply:
Just as my 2 cents: as a maintainer, I have my own fork, and if I need to work with the upstream, I just reference upstream, so I don't have to have two separate repositories cloned locally.